src/toil/wdl/toilwdl.py

```diff
@@ -0,0 +1,2077 @@
+                    fn_section = fn_section + '    except:\n'
+                    fn_section = fn_section + '        ' + job_declaration_name + '_fs = os.path.join(tempDir, ' + job_declaration_name + '["' + job_declaration_key + '"][1])\n'
+                else:
+                    fn_section = fn_section + '    try:\n'
+                    fn_section = fn_section + '        ' + job_declaration_name + '_fs = job.fileStore.readGlobalFile(' + job_declaration_name + '[0], userPath=os.path.join(tempDir, ' + job_declaration_name + '[1]))\n'
+                    fn_section = fn_section + '    except:\n'
+                    fn_section = fn_section + '        ' + job_declaration_name + '_fs = os.path.join(tempDir, ' + job_declaration_name + '[1])\n\n'
+                job_declaration_key = None
+            if job_declaration_type == 'ArrayFile':
+                # these are handled in write_function_cmdvarprep()
+                pass
+
+        return fn_section
+
+    def write_function_bashscriptline(self, job_task_reference, job_alias):
+        '''
+        Writes a function to create a bashscript for injection into the docker
+        container.
+
+        :param job_task_reference: The job referenced in WDL's Task section.
+        :param job_alias: The actual job name to be written.
+        :return: A string writing all of this.
+        '''
+        fn_section = ''
+        fn_section = fn_section + "    generate_docker_bashscript_file(temp_dir=tempDir, docker_dir='/data', globs=["
+        if self.tasks_dictionary[job_task_reference]['outputs']:
+            for output in self.tasks_dictionary[job_task_reference]['outputs']:
+                if output[1] == 'ArrayFile' or 'File':
+                    output_filename = output[2][0]
+                    fn_section = fn_section + "'" + output_filename + "', "
+                else:
+                    raise NotImplementedError
+            if fn_section.endswith(', '):
+                fn_section = fn_section[:-2]
+        fn_section = fn_section + "], cmd=cmd, job_name='" + str(job_alias) + "')\n"
+        fn_section = fn_section + '\n'
+
+        return fn_section
+
+    def write_function_dockercall(self, job_task_reference, docker_image):
+        '''
+        Writes a string containing the apiDockerCall() that will run the job.
+
+        :param job_task_reference: The name of the job calling docker.
+        :param docker_image: The corresponding name of the docker image.
+                                                            e.g. "ubuntu:latest"
+        :return: A string containing the apiDockerCall() that will run the job.
+        '''
+        fn_section = ''
+        fn_section = fn_section + '    apiDockerCall(job, image
```

Comment:
Shouldn't this delete the output_file afterwards? An option to keep it would be useful for debugging.

Reply:
Fixed.